module_utils/ansible_nailgun_cement.py

```diff
@@ -138,3 +162,29 @@ def parse_template_from_file(file_name, module):
     except Exception as e:
         module.fail_json(msg='Error while reading template file: ' + str(e))
     return template_dict
+
+
+def find_organization(module, name, failsafe=False):
+    org = Organization(name=name)
+    response = org.search(set(), {'search': 'name="{}"'.format(name)})
+    return handle_find_response(module, response, message="No organization found for %s" % name, failsafe=failsafe)
+
+
+def find_lifecycle_environment(module, name, organization, failsafe=False):
+    response = entities.LifecycleEnvironment(name=name, organization=organization).search()
+    return handle_find_response(module, response, message="No lifecycle environment found for %s" % name, failsafe=failsafe)
+
+
+def handle_find_response(module, response, message="", failsafe=False):
+    if len(response) == 1:
+        return response[0]
+    elif failsafe:
+        return None
+    else:
+        module.fail_json(msg=message)
+
```

Comment:
I talked with @ehelms about the find functionality. Basically, it seems to make sense to write individual find functions for each entity: Because the API isn't always consistent, a single function that is capable of finding any entity with any input params might become quite complex. Writing a find function per entity will be more flexible in the future, and a change in the API to one model/entity won't affect finding all other entities.

I agree the default message should be implemented. I'll do that.